docs/components/List/List.stories.mdx

```diff
@@ -1,9 +1,175 @@
 import { Meta } from "@storybook/addon-docs";
+import { Card } from "@jobber/components/Card";
+import { List } from "@jobber/components/List";
 
 <Meta title="Components/Lists and Tables/List" />
 
 # List
 
-Displays a collection of objects that are either the same type or related to one
-another, like billing or activity feed. The main purpose of a list is to help
-the user find an object that leads to its full representation.
+The `List` component renders a collection of items in a structured way. It
+supports both [flat lists](#basic-flat-list) and
+[sectioned lists](#sectioned-list), and allows for
+[custom rendering](#custom-rendering-advanced-usage) of list items via a render
+function.
+
+## Design and Usage Guidelines
+
+The `List` component is a great solution if you are looking to display a
+collection of objects that are either the same type or related to one another,
+like billing or activity feed. The main purpose of a list is to help the user
+find an object that leads to its full representation.
+
+The majority of list implementations can be achieved without needing to
+customize rendering. If your items conform to the `ListItemProps` interface, the
+component will render them automatically with the default behavior. A prop of
+type `ListItemProps` must be used when using default rendering.
+
+#### Basic Flat List
+
+<Canvas>
+  <Card>
+    <List
+      items={[
+        {
+          id: 1,
+          content: "Content",
+          icon: "work",
+          iconColor: "interactive",
+          title: "Title",
+          value: "Value",
+          onClick: () => alert("First list item clicked"),
+        },
+        {
+          id: 2,
+          content: "Content",
+          icon: "work",
+          iconColor: "interactive",
+          title: "Title",
+          value: "Value",
+          onClick: () => alert("Second list item clicked"),
+        },
+      ]}
+    />
+  </Card>
+</Canvas>
+
+#### Sectioned List
+
+If a section field is present in any item, the list will group the items by
+their section and display them under section headers.
```

Comment:
one thing that I discovered when playing around with Sectioned Lists is that giving a `section` basically sets a flag saying "I am now in section mode"

since `section` is optional, it's possible to not have a `section` in each of your items.

the currently undocumented behavior that happens in this case is that you get a hard coded header of `Other`

for the existing example I've removed the `section` from something in the middle, and added one more entry at the end that has no `section`

![image](https://github.com/user-attachments/assets/46940ac5-86dd-412e-822c-da77593e20e9)

it'll group everything with the first occurrence of something missing a `section`

I did some digging around in the product and I'm about 98% sure that all the cases where we use a `section` it will never have a missing `section`.

I had no idea it did this and it's kind of debatable if it even should..... but that's another topic. at the very least, let's document the behavior.

maybe something like 

> If even one `section` key exists in any of `items` then it will be expected that every object within that collection will have a `section` defined. If a `section` is missing, "Other" will be used, and any "Other" items will be grouped with that first occurrence that has no `section`.